### DISCUSSION TITLE
Having trouble connecting and getting an IP address after connecting srsRAN, gNB, and srsUE using ZMQ

### DISCUSSION THREAD

**Discussion by aayamrajshakya** [srsran/srsRAN_Project#413] [General Help] [answered]
[ue_zmq.conf.txt](https://github.com/srsran/srsRAN_Project/files/13981082/ue_zmq.conf.txt)
[ue.log](https://github.com/srsran/srsRAN_Project/files/13981083/ue.log)
![Screenshot from 2024-01-18 13-47-11](https://github.com/srsran/srsRAN_Project/assets/144458398/5528f8a7-ba3d-41c3-b62c-19980e91d3dc)
[mme.log](https://github.com/srsran/srsRAN_Project/files/13981084/mme.log)
[gnb_zmq.yaml.txt](https://github.com/srsran/srsRAN_Project/files/13981085/gnb_zmq.yaml.txt)
[gnb.log](https://github.com/srsran/srsRAN_Project/files/13981086/gnb.log)
[amf.yaml.txt](https://github.com/srsran/srsRAN_Project/files/13981087/amf.yaml.txt)
[amf.log](https://github.com/srsran/srsRAN_Project/files/13981088/amf.log)

We tried changing the config files according to the documentation. We still have trouble getting an IP address assigned to the UE. Any ideas where we are messing up? Thank You.

**Comment by pgawlowicz** [comment]
did you create a network namespace for the ue?
```
sudo ip netns add ue1
```

**Reply by aayamrajshakya** [reply]
Yes, i did it

**Comment by pgawlowicz** [comment]
It seems that AMF does not accept the UE, check the logs: 
```
01/17 12:19:55.548: [amf] INFO: InitialUEMessage (../src/amf/ngap-handler.c:401)
01/17 12:19:55.548: [amf] INFO: [Added] Number of gNB-UEs is now 1 (../src/amf/context.c:2550)
01/17 12:19:55.548: [amf] INFO:     RAN_UE_NGAP_ID[0] AMF_UE_NGAP_ID[1] TAC[7] CellID[0x19b0] (../src/amf/ngap-handler.c:562)
01/17 12:19:55.548: [amf] INFO: [suci-0-001-01-0000-0-0-0123456780] Unknown UE by SUCI (../src/amf/context.c:1835)
01/17 12:19:55.548: [amf] INFO: [Added] Number of AMF-UEs is now 1 (../src/amf/context.c:1616)
01/17 12:19:55.548: [gmm] INFO: Registration request (../src/amf/gmm-sm.c:1165)
01/17 12:19:55.548: [gmm] INFO: [suci-0-001-01-0000-0-0-0123456780]    SUCI (../src/amf/gmm-handler.c:166)
01/17 12:19:55.627: [gmm] ERROR: [suci-0-001-01-0000-0-0-0123456780] HTTP response error [500] (../src/amf/gmm-sm.c:1675)
01/17 12:19:55.627: [amf] WARNING: [suci-0-001-01-0000-0-0-0123456780] Registration reject [9] (../src/amf/nas-path.c:219)
01/17 12:19:55.754: [amf] INFO: UE Context Release [Action:3] (../src/amf/ngap-handler.c:1698)
01/17 12:19:55.754: [amf] INFO:     RAN_UE_NGAP_ID[0] AMF_UE_NGAP_ID[1] (../src/amf/ngap-handler.c:1699)
01/17 12:19:55.754: [amf] INFO:     SUCI[suci-0-001-01-0000-0-0-0123456780] (../src/amf/ngap-handler.c:1702)
01/17 12:19:55.754: [amf] INFO: [Removed] Number of gNB-UEs is now 0 (../src/amf/context.c:2557)
01/17 12:19:55.754: [amf] INFO: [Removed] Number of AMF-UEs is now 0 (../src/amf/context.c:1709)
```

did you populate the UE database correctly?

**Reply by pgawlowicz** [reply]
@aayamrajshakya any update on this issue?

**Reply by aayamrajshakya** [reply, marked as answer]
Good morning, Mr. Piotr. Sorry for the late reply, I have been occupied with classworks lately. Yes, I finally figured out the issue; we just missed configuring the _nrf.yaml_ 
![Screenshot_from_2024-01-22_12-26-23 (2)](https://github.com/srsran/srsRAN_Project/assets/144458398/1b5b7014-43f5-4c21-98e4-ab9716ba3ed9)

**Reply by pgawlowicz** [reply]
great! thanks for your feedback!
